firmware/common2015/utils/logger/logger.cpp

```diff
@@ -0,0 +1,66 @@
+#include "logger.hpp"
+
+#include <cstdarg>
+#include <algorithm>
+
+#include "mbed.h"
+
+
+const char* LOG_LEVEL_STRING[] = { FOREACH_LEVEL(GENERATE_STRING) };
+
+
+/* The initial logging level shows startup info along with any
+ * warning messages [but hopefully there's none of those :) ].
+ */
+
+
+/**
+ * Active logging.
+ */
+volatile bool isLogging = RJ_LOGGING_EN;
+
+
+/**
+ * Current log level.
+ */
+volatile uint8_t rjLogLevel;
+
+
+/**
+ * [log The system-wide logging interface function. All log messages go through this.]
+ * @param logLevel [The "importance level" of the called log message.]
```

Comment:
Let's avoid copying the comments for things from headers into cpp files - they'll inevitably get out of sync if anything changes :)